plugins/modules/smart_proxy.py

```diff
@@ -101,13 +111,53 @@ def main():
         foreman_spec=dict(
             name=dict(required=True),
             url=dict(required=True),
+            lifecycle_environments=dict(required=False, type='entity_list'),
             download_policy=dict(required=False, choices=['background', 'immediate', 'on_demand']),
         ),
-        required_plugins=[('katello', ['download_policy'])],
+        required_plugins=[('katello', ['lifecycle_environments', 'download_policy'])],
     )
 
     with module.api_connection():
-        module.run()
+        handle_lifecycle_environments = not module.desired_absent and 'lifecycle_environments' in module.foreman_params
+        if handle_lifecycle_environments:
+            module.lookup_entity('lifecycle_environments')
+            lifecycle_environments = module.foreman_params.pop('lifecycle_environments', [])
+
+        smart_proxy = module.run()
+
+        if handle_lifecycle_environments:
+            if smart_proxy['id'] == -1:
```

Comment:
the `id = -1` this is something that we shouldn't expose in the modules. in other modules we lookup the entity twice and decide on the before/after state essentially:

https://github.com/theforeman/foreman-ansible-modules/blob/a5a7be69678eb7992882f95b12e5de26b7f37027/plugins/modules/role.py#L118-L128

we could introduce a "newly created" property, or something, but the `-1` should remain an implementation detail.

Reply:
I made a fix, is this what you meant?